### DISCUSSION TITLE
Slidesperview auto with slides having width set to auto do not bring all slides in view

### DISCUSSION THREAD

**Discussion by swbhalla** [nolimits4web/swiper#4602] [Q&A]
I have a button component and am applying swiper to a list of such buttons. Button text can be 2-3 words and I want them to occupy their natural width so have applied width auto. Also have configured slidesPerView : 'auto'. But I noticed it does not swipe past 4th button while i have 7 in total. Is there anything i am missing?

// swiper slide width set to auto
.swiper-slide {
   width: auto !important;
}

//slidesperview set to auto
this.swiperConfig = {
      slidesPerView : 'auto',
      nextButton : '.swiper-button-next',
      prevButton : '.swiper-button-prev'
    }

**Comment by Baribj** [comment]
Did you find a solution for this?

I noticed that after my custom font loads, the buttons get a bit wider, by which time Swiper has already calculated the width of the slides, but now they are wider and the calculation are wrong inability to swipe past the fifth button.

Any suggestions?

**Comment by Baribj** [comment]
I dug a bit deeper into the documentations and with a very hacky bit working solution:

```
<Swiper
 grabCursor={true}
          slidesPerView="auto"
          spaceBetween={10}
          freeMode={true}
          modules={[FreeMode]}
          onSwiper={async (swiper) => {
            await document.fonts.ready;
            swiper.update();
          }}
        >
.
.
.
.
.
</Swiper>
```
When onSwiper receives the swipe instance, we check if fonts have been loaded and call swiper.update() after the promise resolves to recalculate the container width.
